src/t8_schemes/t8_standalone/t8_standalone_implementation.hxx

```diff
@@ -0,0 +1,1568 @@
+   *
+   * This is a convenience function, and can be implemented via
+   * \ref t8_element_count_leaves.
+   */
+  static inline t8_gloidx_t
+  count_leaves_from_root (int level)
+  {
+    T8_ASSERT (level >= 0);
+    if constexpr (TEclass == T8_ECLASS_PYRAMID) {
+      SC_ABORT ("Not implemented yet.\n");
+    }
+    return 1LL << (level * T8_ELEMENT_DIM[TEclass]);
+  }
+
+  /** Compare two elements.
+   * \param [in] elem1  The first element.
+   * \param [in] elem2  The second element.
+   * \return       negative if elem1 < elem2, zero if elem1 equals elem2
+   *               and positive if elem1 > elem2.
+   *  If elem2 is a copy of elem1 then the elements are equal.
+   */
+  static inline int
+  element_compare (const t8_element_t *elem1, const t8_element_t *elem2)
+  {
+    T8_ASSERT (t8_standalone_scheme<TEclass>::element_is_valid (elem1));
+    T8_ASSERT (t8_standalone_scheme<TEclass>::element_is_valid (elem2));
+
+    const t8_standalone_element<TEclass> *e1 = (const t8_standalone_element<TEclass> *) elem1;
+    const t8_standalone_element<TEclass> *e2 = (const t8_standalone_element<TEclass> *) elem2;
+
+    const int maxlvl = SC_MAX (e1->level, e2->level);
+
+    const t8_linearidx_t id1 = t8_standalone_scheme<TEclass>::element_get_linear_id ((const t8_element_t *) e1, maxlvl);
+    const t8_linearidx_t id2 = t8_standalone_scheme<TEclass>::element_get_linear_id ((const t8_element_t *) e2, maxlvl);
+    if (id1 == id2) {
+      if (e1->level == e2->level) {
+        return 0;
+      }
+      else {
+        return e1->level - e2->level;
+      }
+    }
+    return id1 < id2 ? -1 : 1;
+  }
+
+  // ################################################____VISUALIZATION____################################################
+
+  /** Compute the coordinates of a given element vertex inside a reference tree
+   *  that is embedded into [0,1]^d (d = dimension).
+   *   \param [in] elem      The element to be considered.
+   *   \param [in] vertex The id of the vertex whose coordinates shall be computed.
+   *   \param [out] coords An array of at least as many doubles as the element's dimension
+   *                      whose entries will be filled with the coordinates of \a vertex.
+   */
+  static inline void
+  element_get_vertex_reference_coords (const t8_element_t *elem, const int vertex, double coords[])
+  {
+    T8_ASSERT (t8_standalone_scheme<TEclass>::element_is_valid (elem));
+
+    const t8_standalone_element<TEclass> *el = (const t8_standalone_element<TEclass> *) elem;
+    if constexpr (TEclass == T8_ECLASS_VERTEX) {
+      return;
+    }
+    else {
+      int coords_int[T8_ELEMENT_DIM[TEclass]];
+      T8_ASSERT (0 <= vertex && vertex < T8_ELEMENT_NUM_CORNERS[TEclass]);
+      t8_standalone_scheme<TEclass>::element_compute_coords (el, vertex, coords_int);
+      for (int i = 0; i < T8_ELEMENT_DIM[TEclass]; i++) {
+        coords[i] = coords_int[i] / (double) t8_standalone_scheme<TEclass>::get_root_len ();
```

Comment:
```suggestion
        coords[i] = (double) coords_int[i] / t8_standalone_scheme<TEclass>::get_root_len ();
```
Please double check this befor adding. But I think dividing and int by a double results into a double (again: not 100% sure, but please look it up). 

also use a more descriptive name than "i".

Reply:
Just had another look at it. It shouldn't matter where you put the cast to a double. As long as one of the numbers is a double you will use the / operator of a double.